### ISSUE TITLE
Spanish translation

### ISSUE THREAD

**Issue by KD4bra** [hackerncoder/butterfly-soup_translations#1]
Hi! I'm the KD4bra guy from itch.io
I'm a spanish native, so I could help with this.

**Comment by hackerncoder** [comment]
Translation is done in the translation files (empty versions are found under translations/general). You will need to check arial.ttf if it has all the needed characters for spanish (you can find it in butterfly soup/game/ or wait until I upload a copy)

**Comment by hackerncoder** [comment]
I have added a folder for spanish translations, and uploaded arial.ttf, as I said in the comment above, check if arial.ttf contains all necessary characters for spanish, and then begin translating!

**Comment by KD4bra** [comment]
Thanks! Yes, the font has all the necessary characters.
I'll start working on the translation.

**Comment by hackerncoder** [comment]
I was thinking of keeping the issue open, until the translation is finished. Useful for keeping things in one place.

**Comment by hackerncoder** [comment]
I had to change "general" to "spanish" in all the translation files (else it won't work in game), so you will need to update your files @KD4bra.

**Comment by hackerncoder** [comment]
How would you translate "the webpage is in english" or something similar? 

Context: Telling people that the self-voicing page is in english, when translating "Toggles assistive {a=https://www.renpy.org/l/voicing}self-voicing{/a}. " (Line 402 in screens.rpy)

**Comment by KD4bra** [comment]
I added a general disclaimer for the languages of the webpages since it's the same for almost every page.

Also, Ren'Py gives `Argument 's' has incorrect type (expected unicode, got str)` in line 37, tl\spanish\1_diya.rpy
Apparently the problem is the `{w=[time]}` command since taking them out runs normally, yet in other translations the command is present and doesn't give the error.
Could be a problem with my Ren'Py installation? It references renpy\text\text.py in the traceback:

> While running game code:
  File "game/tl/spanish/1_diya.rpy", line 37, in script
    "Aun así, {w=0.35} ella era una adorable princesa. Y trágicamente, estaba destinada a pincharse el dedo en una aguja y morir."
TypeError: Argument 's' has incorrect type (expected unicode, got str)
 -- Full Traceback ------------------------------------------------------------
 Full traceback:
File "game/tl/spanish/1_diya.rpy", line 37, in script
    "Aun así, {w=0.35} ella era una adorable princesa. Y trágicamente, estaba destinada a pincharse el dedo en una aguja y morir."
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\ast.py", line 708, in execute
    renpy.exports.say(who, what, *args, **kwargs)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\exports.py", line 1345, in say
    who(what, *args, **kwargs)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\character.py", line 1142, in __call__
    self.do_display(who, what, cb_args=self.cb_args, **display_args)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\character.py", line 842, in do_display
    **display_args)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\character.py", line 591, in display_say
    rv = renpy.ui.interact(mouse='say', type=type, roll_forward=roll_forward)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\ui.py", line 297, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\core.py", line 2702, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, **kwargs)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\core.py", line 3194, in interact_core
    self.draw_screen(root_widget, fullscreen_video, (not fullscreen_video) or video_frame_drawn)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\core.py", line 2094, in draw_screen
    renpy.config.screen_height,
  File "renpy/display/render.pyx", line 490, in renpy.display.render.render_screen (gen\renpy.display.render.c:6805)
    rv = render(root, width, height, st, st)
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\layout.py", line 722, in render
    surf = render(child, width, height, cst, cat)
  File "renpy/display/render.pyx", line 166, in renpy.display.render.render (gen\renpy.display.render.c:4075)
    cpdef render(d, object widtho, object heighto, double st, double at):
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\layout.py", line 722, in render
    surf = render(child, width, height, cst, cat)
  File "renpy/display/render.pyx", line 166, in renpy.display.render.render (gen\renpy.display.render.c:4075)
    cpdef render(d, object widtho, object heighto, double st, double at):
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\layout.py", line 722, in render
    surf = render(child, width, height, cst, cat)
  File "renpy/display/render.pyx", line 166, in renpy.display.render.render (gen\renpy.display.render.c:4075)
    cpdef render(d, object widtho, object heighto, double st, double at):
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\screen.py", line 675, in render
    child = renpy.display.render.render(self.child, w, h, st, at)
  File "renpy/display/render.pyx", line 166, in renpy.display.render.render (gen\renpy.display.render.c:4075)
    cpdef render(d, object widtho, object heighto, double st, double at):
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\layout.py", line 722, in render
    surf = render(child, width, height, cst, cat)
  File "renpy/display/render.pyx", line 166, in renpy.display.render.render (gen\renpy.display.render.c:4075)
    cpdef render(d, object widtho, object heighto, double st, double at):
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\display\layout.py", line 1127, in render
    st, at)
  File "renpy/display/render.pyx", line 166, in renpy.display.render.render (gen\renpy.display.render.c:4075)
    cpdef render(d, object widtho, object heighto, double st, double at):
  File "renpy/display/render.pyx", line 259, in renpy.display.render.render (gen\renpy.display.render.c:3520)
    rv = d.render(widtho, heighto, st, at)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\text\text.py", line 1996, in render
    virtual_layout = Layout(self, width, height, renders, drawable_res=False, size_only=True)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\text\text.py", line 612, in __init__
    p, rtl = self.rtl_paragraph(p)
  File "D:\Instaladores-Programas\renpy-7.3.5-sdk\renpy\text\text.py", line 1205, in rtl_paragraph
    s, direction = log2vis(s, direction)
TypeError: Argument 's' has incorrect type (expected unicode, got str)

**Comment by hackerncoder** [comment]
I had the same error running my android build, using Renpy 7.4 fixed it, so ~~if you aren't (I can see from the log you are using 7.3.5)~~ try using it.

**Comment by hackerncoder** [comment]
Also, you have to keep your local files up to date with those on github, I have rewritten spanish/screens.rpy compared to back when I originally created the translation files, and as such the translation file tl/screens.rpy had lines that aren't needed anymore

**Comment by KD4bra** [comment]
Yeah, my Ren'Py version was the problem. Now it displays normally.

**Comment by N0th1ng2Do** [comment]
Hello, as i've seen this hasnt been worked on for a while i decided to take on the project, im a spanish native too and i have been already doing some progress on the weblate site (under the username 7Falcon7), if anything is needed just tell me, cheers!

**Comment by hackerncoder** [comment]
Thank you @N0th1ng2Do !

**Comment by N0th1ng2Do** [comment]
Also, is there any way i can throw the .po file into the game as a better way to see how its going/quickly check for any errors?
Im kind of a newbie at Ren'Py (and GitHub too lol) so im getting kinda lost with all of it

**Comment by hackerncoder** [comment]
No.

You need to turn the PO into Ren'Py translation, then put that into the game. I think I have something at https://queerscriptors.org/developer

**Comment by N0th1ng2Do** [comment]
Ok i managed to get it to work, thank you

**Comment by N0th1ng2Do** [comment]
man its been a while, i havent been able to do much but ill still try to progress on it, ive encountered a few font errors with spanish's special characters so hopefully you could help me with those, thank you
